arch/riscv/src/lib.rs

```diff
@@ -45,105 +45,101 @@ extern "C" {
+        .option push
+        .option norelax
+
+        la gp, {gp}                 // Set the global pointer from linker script.
+
+        // Re-enable linker relaxations.
+        .option pop
+
+        // Initialize the stack pointer register. This comes directly from
+        // the linker script.
+        la sp, {estack}             // Set the initial stack pointer.
+
+        // Set s0 (the frame pointer) to the start of the stack.
+        add  s0, sp, zero           // s0 = sp
+
+        // Initialize mscratch to 0 so that we know that we are currently
+        // in the kernel. This is used for the check in the trap handler.
+        csrw 0x340, zero            // CSR=0x340=mscratch
+
+        // INITIALIZE MEMORY
+
+        // Start by initializing .bss memory. The Tock linker script defines
+        // `_szero` and `_ezero` to mark the .bss segment.
+        la a0, {sbss}               // a0 = first address of .bss
+        la a1, {ebss}               // a1 = first address after .bss
+
+      100: // bss_init_loop
+        beq  a0, a1, 101f           // If a0 == a1, we are done.
+        sw   zero, 0(a0)            // *a0 = 0. Write 0 to the memory location in a0.
+        addi a0, a0, 4              // a0 = a0 + 4. Increment pointer to next word.
+        j 100b                      // Continue the loop.
+
+      101: // bss_init_done
+
+
+        // Now initialize .data memory. This involves coping the values right at the
+        // end of the .text section (in flash) into the .data section (in RAM).
+        la a0, {sdata}              // a0 = first address of data section in RAM
+        la a1, {edata}              // a1 = first address after data section in RAM
+        la a2, {etext}              // a2 = address of stored data initial values
+
+      200: // data_init_loop
+        beq  a0, a1, 201f           // If we have reached the end of the .data
+                                    // section then we are done.
+        lw   a3, 0(a2)              // a3 = *a2. Load value from initial values into a3.
+        sw   a3, 0(a0)              // *a0 = a3. Store initial value into
+                                    // next place in .data.
+        addi a0, a0, 4              // a0 = a0 + 4. Increment to next word in memory.
+        addi a2, a2, 4              // a2 = a2 + 4. Increment to next word in flash.
+        j 200b                      // Continue the loop.
+
+      201: // data_init_done
+
+        // With that initial setup out of the way, we now branch to the main
+        // code, likely defined in a board's main.rs.
+        j main
+    ",
+        gp = sym __global_pointer,
+        estack = sym _estack,
+        sbss = sym _szero,
+        ebss = sym _ezero,
+        sdata = sym _srelocate,
+        edata = sym _erelocate,
+        etext = sym _etext,
+    );
 }
 
-#[cfg(any(doc, all(target_arch = "riscv32", target_os = "none")))]
-core::arch::global_asm!("
-            .section .riscv.start, \"ax\"
```

Comment:
This change drops the "[a]llocate, e[x]ecutable" flags from the section label here. I'm not sure it's critical since the linker script should really control this anyway (and does somewhat implicitly by being included in `.text`), but I'm just raising this here in case there's a RISC-V nuance that made these important not to lose.

Reply:
I think this was just copied and pasted from somewhere at some time and isn't needed.